pkg/yang/identity.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package yang
+
+import (
+	"fmt"
+	"sync"
+)
+
+// This file implements data structures and functions that relate to the
+// identity type.
+
+type identityDictionary struct {
+	mu   sync.Mutex
+	dict map[string]*resolvedIdentity
```

Comment:
Only make this a reference if you needto.  The type is small so it is more efficient to just be a map of resolvedIdentity rather than a reference to one.